apps/tokenization/src/agreement/agreement.controller.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Logger,
+  Param,
+  Post,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import {
+  ApiBody,
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { AgreementDTO, AgreementService, BlockchainPropertiesService, CreateAgreementDTO } from '@zero-labs/tokenization-api';
+import { providers, Wallet } from 'ethers';
+import { signAgreement } from '@zero-labs/tokenization';
+import { AccountService } from '../account/account.service';
+
+@ApiTags('agreement')
+@Controller('agreement')
+@UsePipes(ValidationPipe)
+export class AgreementController {
+  private readonly logger = new Logger(AgreementController.name, {
+    timestamp: true,
+  });
+
+  constructor(
+    private readonly blockchainPropertiesService: BlockchainPropertiesService,
+    private readonly agreementService: AgreementService,
+    private readonly accountService: AccountService
+  ) {}
+
+  @Post()
+  @ApiBody({ type: CreateAgreementDTO })
+  @ApiCreatedResponse({
+    type: AgreementDTO,
+    description: 'Agreement data',
+  })
+  public async create(
```

Comment:
shouldn't we protect these endpoints by guards ?

Reply:
yup, thanks for pointing that out

Reply:
Addressed it now - marking as resolved.